blockchain/filedao/staging_buffer.go

```diff
@@ -36,22 +36,19 @@ func (s *stagingBuffer) Get(pos uint64) (*block.Store, error) {
 	return s.buffer[pos], nil
 }
 
-func (s *stagingBuffer) Put(pos uint64, blkBytes []byte) (bool, error) {
+func (s *stagingBuffer) Put(pos uint64, blk *block.Store) (bool, error) {
 	if pos >= s.size {
 		return false, ErrNotSupported
 	}
-	blk, err := s.deser.DeserializeBlockStore(blkBytes)
-	if err != nil {
-		return false, err
-	}
 	s.buffer[pos] = blk
 	return pos == s.size-1, nil
 }
 
 func (s *stagingBuffer) Serialize() ([]byte, error) {
 	blkStores := []*iotextypes.BlockStore{}
+	// blob sidecar data are stored separately
 	for _, v := range s.buffer {
-		blkStores = append(blkStores, v.ToProto())
+		blkStores = append(blkStores, v.ToProtoWithoutSidecar())
```

Comment:
here is where 16 blocks are packed together and written to file/disk, so need to remove blob sidecar before doing that